### PR TITLE
Harfbuzz support for KLEE.

### PR DESCRIPTION
**Pull request by rsprabery**
- Removed glib dependency in harfbuzz
- Remove cairo dependency in harfbuzz
- Explicitly build hb-ucdn.la to work around brittle build system

### CODE REVIEW

File: fuzzers/klee/fuzzer.py [hunk 1]
```diff
@@ -353,10 +441,19 @@ def fuzz(input_corpus, output_corpus, target_binary):
...
+        '-max-solver-time',
+        '30s',
+        '-log-timed-out-queries',
+        '--max-time',
+        '{}s'.format(seconds),
+        '-libc',
+        'uclibc',
+        '-libcxx',
+        '-posix-runtime',
+        '--disable-verify',  # Needed because debug builds don't always work.
```
Comment (laurentsimon) [inline fuzzers/klee/fuzzer.py:454]: shall we add this only for this target?

File: fuzzers/klee/fuzzer.py [hunk 2]
```diff
@@ -124,20 +140,89 @@ def get_bc_files():
...
+            return [os.path.join(os.environ['OUT'], fuzzname)]
+
+    # For the reamining oss-projects, use some heuristics.
+    # We look for binaries in the OUT directory and take it as our targets.
+    # Note that we may return multiple binaries: this is necessary because
+    # sometimes multiple binaries are generated and we don't know which will
+    # be used for fuzzing (e.g., zlib benchmark).
+    # dictionary above.
     out_dir = os.environ['OUT']
-    if is_benchmark('sqlite3'):
```
Comment (laurentsimon) [inline fuzzers/klee/fuzzer.py:129]: is sqlite3 removed entirely? I think it was integrated.

File: fuzzers/klee/fuzzer.py [hunk 3]
```diff
@@ -124,20 +140,89 @@ def get_bc_files():
...
+    print("[get_fuzz_target] targets: %s" % fuzz_targets)
+    return fuzz_targets
 
-    # For non oss-projects, FUZZ_TARGET contain the target binary
-    fuzz_target = os.getenv('FUZZ_TARGET', None)
-    if fuzz_target is not None:
-        return fuzz_target
 
-    raise ValueError('Cannot determine fuzz target')
+def fix_fuzzer_lib():
```
Comment (laurentsimon) [inline fuzzers/klee/fuzzer.py:183]: strange that there is so much changed for below lines. This was already in the code, right?

**Comment by rsprabery** [comment]
Closing this as updating to master required some additional work. 

This harfbuzz work and the rest of the KLEE changes are being committed in https://github.com/google/fuzzbench/pull/645

**Pull request closed without merging**